infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -197,6 +200,26 @@ public void apply(LogReplicationEntry message) {
         recvSeq++;
     }
 
+    /**
+     * Create a shadow table per stream to replicate.
+     *
+     * A shadow stream is the combination of the stream it shadows and the snapshot sync request Id,
+     * in order to make them unique between snapshot syncs.
+     *
+     * @param newSnapshotRequestId
+     */
+    private void createShadowTables(UUID newSnapshotRequestId) {
+        for (String streamName : config.getStreamsToReplicate()) {
+            UUID streamId = CorfuRuntime.getStreamID(streamName);
+            // Remove mapping from previous shadow stream to avoid memory leak
+            uuidMap.values().removeIf(regularStream -> regularStream.equals(streamId));
+            UUID shadowStreamId = new UUID(streamId.getMostSignificantBits(), newSnapshotRequestId.getLeastSignificantBits());
+            uuidMap.put(streamId, shadowStreamId);
+            uuidMap.put(shadowStreamId, streamId);
+            streamViewMap.put(streamId, streamName);
```

Comment:
streamViewMap is init in the constructor and should not be changed later, right?